core/internal/filetransfer/file_transfers.go

```diff
@@ -22,17 +24,20 @@ func NewFileTransfers(
 	logger *observability.CoreLogger,
 	fileTransferStats FileTransferStats,
 ) *FileTransfers {
-	defaultFileTransfer := &DefaultFileTransfer{
-		logger:            logger,
-		client:            client,
-		fileTransferStats: fileTransferStats,
-	}
+	defaultFileTransfer := NewDefaultFileTransfer(client, logger, fileTransferStats)
+	gcReferenceFileTransfer := NewGCSFileTransfer(nil, logger, fileTransferStats)
 	return &FileTransfers{
-		Default: defaultFileTransfer,
+		Default:      defaultFileTransfer,
+		GCSReference: gcReferenceFileTransfer,
 	}
 }
 
 // Returns the appropriate fileTransfer depending on task
 func (ft *FileTransfers) GetFileTransferForTask(task *Task) FileTransfer {
-	return ft.Default
+	switch {
+	case ft.GCSReference.CanHandle(task):
+		return ft.GCSReference
```

Comment:
we shouldn't ever be reaching this until we enable core for ref artifacts, but I can also comment it out for now if we want and just return `ft.Default`